sdk/src/userMap/userMap.ts

```diff
@@ -303,65 +303,59 @@ export class UserMap implements UserMapInterface {
 				},
 			];
 
-			const rpcJSONResponse: any =
-				// @ts-ignore
-				await this.connection._rpcRequest('getProgramAccounts', rpcRequestArgs);
-
-			const rpcResponseAndContext: RpcResponseAndContext<
-				Array<{
-					pubkey: PublicKey;
-					account: {
-						data: [string, string];
-					};
-				}>
-			> = rpcJSONResponse.result;
-
+			// @ts-ignore
+			const rpcJSONResponse: any = await this.connection._rpcRequest('getProgramAccounts', rpcRequestArgs);
+			const rpcResponseAndContext: RpcResponseAndContext<Array<{ pubkey: PublicKey; account: { data: [string, string]; } }>> = rpcJSONResponse.result;
 			const slot = rpcResponseAndContext.context.slot;
-
 			this.updateLatestSlot(slot);
 
 			const programAccountBufferMap = new Map<string, Buffer>();
-			for (const programAccount of rpcResponseAndContext.value) {
+			rpcResponseAndContext.value.forEach(programAccount => {
 				programAccountBufferMap.set(
 					programAccount.pubkey.toString(),
 					// @ts-ignore
-					Buffer.from(
-						programAccount.account.data[0],
-						programAccount.account.data[1]
-					)
+					Buffer.from(programAccount.account.data[0], programAccount.account.data[1])
 				);
-			}
+			});
+
+			const concurrencyLimit = 20;
+			const semaphore = new Array(concurrencyLimit).fill(Promise.resolve());
 
-			for (const [key, buffer] of programAccountBufferMap.entries()) {
+			const processAccount = async (key: string, buffer: Buffer) => {
 				if (!this.has(key)) {
 					const userAccount = this.decode('User', buffer);
 					await this.addPubkey(new PublicKey(key), userAccount);
-					this.userMap.get(key).accountSubscriber.updateData(userAccount, slot);
+					this.userMap.get(key)?.accountSubscriber.updateData(userAccount, slot);
 				} else {
 					const userAccount = this.decode('User', buffer);
-					this.userMap.get(key).accountSubscriber.updateData(userAccount, slot);
+					this.userMap.get(key)?.accountSubscriber.updateData(userAccount, slot);
 				}
-				// give event loop a chance to breathe
-				await new Promise((resolve) => setTimeout(resolve, 0));
-			}
+			};
+
+			const promises = Array.from(programAccountBufferMap.entries()).map(async ([key, buffer]) => {
+				const index = await Promise.race(semaphore.map((p, index) => p.then(() => index)));
+				semaphore[index] = processAccount(key, buffer).then(() => {
+					return;
+				});
+			});
+
+			await Promise.all(promises.concat(semaphore));
```

Comment:
```suggestion
			const promises = Array<Promise<void>>();
			programAccountBufferMap.forEach(async (buffer, key) => {
				promises.push(processAccount(key, buffer));
			});
			await Promise.all(promises);
```

you should be able to reduce it to this, not sure if semaphore is necssary